src/mol-plugin-state/animation/built-in/state-snapshots.ts

```diff
@@ -4,23 +4,30 @@
  * @author David Sehnal <david.sehnal@gmail.com>
  */
 
+import { Canvas3DParams } from '../../../mol-canvas3d/canvas3d';
 import { PluginContext } from '../../../mol-plugin/context';
 import { PluginState } from '../../../mol-plugin/state';
+import { ParamDefinition as PD } from '../../../mol-util/param-definition';
 import { PluginStateSnapshotManager } from '../../manager/snapshots';
 import { PluginStateAnimation } from '../model';
 
+
 async function setPartialSnapshot(plugin: PluginContext, entry: Partial<PluginStateSnapshotManager.Entry['snapshot']>, first = false) {
     if (entry.data) {
         await plugin.runTask(plugin.state.data.setSnapshot(entry.data));
         // update the canvas3d trackball with the snapshot
-        if (entry.canvas3d?.props?.trackball) {
-            plugin.canvas3d?.setProps({
-                trackball: entry.canvas3d?.props?.trackball
-            });
-        }
-
     }
 
+    if (entry.canvas3d?.props) {
+        const settings = PD.normalizeParams(Canvas3DParams, entry.canvas3d.props, 'children');
```

Comment:
Could you benchmark the if condition code that normalizes and sets the props? If it takes say <1us I think it might be fine to include... otherwise we should optimize. (And even 1ms might be too long for essentially just changing background color).